src/utils.jl

```diff
@@ -118,4 +118,11 @@ function householderbasis(n)
       u, v = v, u
   end
   u, v
-end
\ No newline at end of file
+end
+
+"""
+    mayberound(λ, x, tol)
+
+Round `λ` to `x` if it is within the tolerance `tol`.
+"""
+mayberound(λ, x, tol) = isapprox(λ, x, atol = tol) ? x : λ
```

Comment:
I think the argument `tol` should be renamed to `atol` to reflect the exact kind of tolerance that is being considered here. Also, it could default to `atol(T)` if you annotate lambda::T in the function signature.

Reply:
good idea, committed ;)